connectors/src/connectors/slack/temporal/activities.ts

```diff
@@ -0,0 +1,479 @@
+  connectorId: string,
+  slackClient: WebClient
+) {
+  return (
+    await Promise.all(
+      messages.map(async (message) => {
+        const text = await processMessageForMentions(
+          message.text as string,
+          connectorId,
+          slackClient
+        );
+
+        const userName = await getUserName(
+          message.user as string,
+          connectorId,
+          slackClient
+        );
+        const messageDate = new Date(parseInt(message.ts as string, 10) * 1000);
+        const messageDateStr = formatDateForUpsert(messageDate);
+
+        return `>> @${userName} [${messageDateStr}]:\n ${text}\n`;
+      })
+    )
+  ).join("\n");
+}
+
+async function upsertToDatasource(
+  dataSourceConfig: DataSourceConfig,
+  documentId: string,
+  documentContent: string,
+  sourceUrl?: string,
+  createAt?: number,
+  retries = 1,
+  waitBetweenRetries = 1000
+) {
+  if (retries < 1) {
+    throw new Error("retries must be greater than 0");
+  }
+  const errors = [];
+  for (let i = 0; i < retries; i++) {
+    try {
+      const res = await _upsertToDatasource(
+        dataSourceConfig,
+        documentId,
+        documentContent,
+        sourceUrl,
+        createAt
+      );
+      return res;
+    } catch (e) {
+      errors.push(e);
+      await new Promise((resolve) => setTimeout(resolve, waitBetweenRetries));
+    }
+  }
+  throw new Error(errors.join("\n"));
+}
+
+async function _upsertToDatasource(
+  dataSourceConfig: DataSourceConfig,
+  documentId: string,
+  documentContent: string,
+  sourceUrl?: string,
+  createAt?: number
+) {
+  const dust_url = `${FRONT_API}/api/v1/w/${dataSourceConfig.workspaceId}/data_sources/${dataSourceConfig.dataSourceName}/documents/${documentId}`;
+  const dust_request_payload = {
+    text: documentContent,
+    sourceUrl: sourceUrl,
+    timestamp: createAt,
+  };
+  const dust_request_config: AxiosRequestConfig = {
+    headers: {
+      Authorization: `Bearer ${dataSourceConfig.workspaceAPIKey}`,
+    },
+    timeout: NETWORK_REQUEST_TIMEOUT_MS,
+  };
+  const dust_request_result = await axios.post(
+    dust_url,
+    dust_request_payload,
+    dust_request_config
+  );
+  if (dust_request_result.status >= 200 && dust_request_result.status < 300) {
+    return;
+  } else {
+    throw new Error(
+      `Failed to upsert to Dust data source ${dataSourceConfig.dataSourceName}: ${dust_request_result.status} ${dust_request_result.statusText}`
+    );
+  }
+}
+
+export async function fetchUsers(
+  slackAccessToken: string,
+  connectorId: string
+) {
+  let cursor: string | undefined;
+  const client = getSlackClient(slackAccessToken);
+  do {
+    const res = await client.users.list({});
+    if (res.error) {
+      throw new Error(`Failed to fetch users: ${res.error}`);
+    }
+    if (!res.members) {
+      throw new Error(`Failed to fetch users: members is undefined`);
+    }
+    for (const member of res.members) {
+      if (member.id && member.real_name) {
```

Comment:
Should we use real_name? Is there a handle? Currently with real_name, is there an "@" in front of the real_name in our logs?

Reply:
Possible to share an example processed chat to convince ourselves that we have the right format

Reply:
Looks great.

- we should probably add an @ before mentions in the messages bodies.
- I'm wondering if real_name is better than handle? Should we use handles? Should we show both?

If it's not too hard I would recommend switching to handles so that it matches what humans see.

cc @fontanierh @gabhubert thoughts?

Reply:
> we should probably add an @ before mentions in the messages bodies.

➕ "Aric Lasry left a set of comments." is semantically quite different from "@Aric Lasry left a set of comments." (for humans, no idea what `text-embedding-ada-002` would say about it)

Reply:
Updated. Let me know what you think guys.
(also fixed the format according to @gabhubert comments [on Slack](https://dust-labs.slack.com/archives/C050A0S2Z7F/p1682673581857599))

![Uploading Screen Shot 2023-04-28 at 12.04.22.png…]()

Reply:
And the text version if you need it:
```
>> @aric [20230411 18:51]:
Here is the design doc to open the discussion about the implementation of the managed data sources:
<https://docs.google.com/document/d/1MjkIJrG2kqZeY0Az_tBRfycmds6TEVTNfjJfrcyjbL4/edit?usp=sharing>

>> @spolu [20230411 19:07]:
Looks great. Will make a pass of comments hopefully tonight with thoughts on your questions.

My main question is how much state we’ll need to maintain to successfully run incremental updates across services and what should that state look like?

>> @aric [20230411 19:10]:
My initial analysis on that is: it will probably be on a per service basis. 
I can do a pass on 2/3 services to get a concrete answer.

>> @spolu [20230411 19:33]:
I think that would be an interesting analysis as ideally we find a state format that is generic?

>> @spolu [20230411 19:34]:
Or maybe not but it’s be in favour of having that leave in its own separate DB. To be discussed in any case :+1: :+1: 

>> @spolu [20230412 11:04]:
@aric left a set of comments. Looks great. Left a few questions that should be easy to address :+1:

>> @aric [20230412 16:27]:
@spolu I think I can have a cool “dirty” PR tonight that will give us a good idea of all the areas of the code base that will need to change. Super excited.

>> @spolu [20230412 16:28]:
Excellent!!

>> @spolu [20230412 16:28]:
:fire:
```